e2e/tests/omnibus-arbitrum-mainnet.toml/Liquidation_Rewards_Registrations.e2e.js

```diff
@@ -0,0 +1,170 @@
+      if (key.startsWith('invoke.')) {
+        const [, artifactName] = key.split('.');
+        const events = value?.artifacts?.txns?.[artifactName]?.events?.RewardsDistributorRegistered;
+        if (events) {
+          for (const event of events) {
+            const [poolId, collateralType, distributor] = event.args;
+            registeredDistributors.push({ poolId, collateralType, distributor });
+          }
+        }
+      }
+    }
+    log({ registeredDistributors });
+  });
+
+  it('should collect all expected distributors for perps liquidations', async () => {
+    const supportedCollateralsIds = await contractRead({
+      wallet,
+      contract: 'PerpsMarketProxy',
+      func: 'getSupportedCollaterals',
+      args: [],
+    });
+    log({ supportedCollateralsIds });
+
+    const systemToken = require('../../deployments/systemToken.json');
+    const synthTokens = require('../../deployments/synthTokens.json');
+
+    // We can only use synth tokens and system token as collaterals
+    // We do NOT need a rewards distributor for system token
+    const supportedCollaterals = supportedCollateralsIds
+      .map((id) => {
+        const synthMarketId = id.toString();
+        if (synthMarketId === '0') {
+          return {
+            synthMarketId,
+            ...systemToken,
+          };
+        }
+        return synthTokens.find((token) => token.synthMarketId === synthMarketId);
+      })
+      .filter(Boolean);
+    log({ supportedCollaterals });
+
+    const liquidatableCollaterals = supportedCollaterals
+      // We do not need rewards distributor for system token
+      .filter((collateral) => collateral.address !== systemToken.address);
+
+    expectedDistributors = [];
+    for (const liquidatableCollateral of liquidatableCollaterals) {
+      const { distributor, poolDelegatedCollateralTypes } = await contractRead({
+        wallet,
+        contract: 'PerpsMarketProxy',
+        func: 'getRegisteredDistributor',
+        args: [liquidatableCollateral.synthMarketId],
+      });
+      for (const collateralType of poolDelegatedCollateralTypes) {
+        expectedDistributors.push({
+          poolId,
+          collateralType,
+          distributor,
+        });
+      }
+    }
+    log({ expectedDistributors });
+  });
+
+  it('should ensure each perps collateral has rewards distributor registered', async () => {
+    for (const expectedDistributor of expectedDistributors) {
+      const registeredDistributor = registeredDistributors.find(
+        ({ poolId, collateralType, distributor }) =>
+          collateralType === expectedDistributor.collateralType &&
+          distributor === expectedDistributor.distributor &&
+          poolId === expectedDistributor.poolId
+      );
+      log({ expectedDistributor, registeredDistributor });
+      assert.ok(
+        registeredDistributor,
+        'Every expected rewards distributor for liquidations is registered'
+      );
+    }
+  });
```

Comment:
This test verifies that cannon state includes all the rewards distributors and we did not "sideload" any transactions without github-based release flow.